pkgs/development/libraries/xdg-desktop-portal/default.nix

```diff
@@ -46,18 +50,31 @@ stdenv.mkDerivation (finalAttrs: {
       substitute "${flatpak.icon-validator-patch}" "$out" \
         --replace "/icon-validator/validate-icon.c" "/src/validate-icon.c"
     '')
+
+    # Allow installing installed tests to a separate output.
+    ./installed-tests-path.patch
+
+    # `XDG_DESKTOP_PORTAL_DIR` originally was used for upstream tests. But we are making use
+    # of this in the NixOS module, this actually blocks any configs from being loaded since
+    # configs are not expected to be placed in a portal implementation or even under the
+    # `share/xdg-desktop-portal/portals/` path.
```

Comment:
I am starting to think we should just avoid the `XDG_DESKTOP_PORTAL_DIR` environment variable and go back to a custom one (as introduced in ec80c5e4c48df8264e5e230bb81eb55e045382dc). We do this with other test-specific environment variables like `GSETTINGS_SCHEMA_DIR`×`NIX_GSETTINGS_OVERRIDES_DIR` as well. We could even point it to `/run/current-system/share/xdg-desktop-portal` and drop our custom joining, it is all global anyway.

Reply:
Something like this might work https://github.com/NixOS/nixpkgs/pull/268197 but this can probably wait after branch-off.

Reply:
https://github.com/NixOS/nixpkgs/pull/268197 sounds good. I originally thought `XDG_DESKTOP_PORTAL_DIR` is used in hyprland upstream packaging but another look suggests it is [not even used](https://github.com/hyprwm/Hyprland/pull/3077/files#diff-2ce94560f72610491d57960707711937fae495ecb2c81a548125676928296220) anymore.